[libcxx] Run tests on Windows/arm64 too

mstorsjo opened the pull request:
Github Actions has got free runners with Windows on ARM64 these days, which can be used for running CI, just as well as the other existing cases.

This qualifies this configuration as a supported target platform, thus add it to the docs in the listing of supported platforms.

mstorsjo committed:
[libcxx] Run tests on Windows/arm64 too

Github Actions has got free runners with Windows on ARM64 these
days, which can be used for running CI, just as well as the other
existing cases.

This qualifies this configuration as a supported target platform,
thus add it to the docs in the listing of supported platforms.

philnik777 commented:
In general I'm happy with this, but I don't really know the GH CI stuff. Maybe @boomanaiden154 could take a look at that.

mstorsjo committed:
Remove the separate config name, as it was identical

Also fix the default runner image name, don't use "windows-latest"
but keep using "windows-2022" like we did before.

philnik777 reviewed (approved):
The libc++ side LGTM, but I'd like someone who knows the GH CI stuff to review.

boomanaiden154 commented:
It's a bit annoying that there are no supported Windows Server SKUs on ARM to keep everything consistent, but that seems to mostly involve waiting on Microsoft to release one.
